### PR TITLE
Use generic currency symbol

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue https://github.com/woocommerce/woocommerce-android/issues/4553

### Description

This PR removes hardcoded "$" currency symbol and uses a utility class to get a currency symbol at runtime.

It also removes CurrencyModule as it isn't necessary - we can use @Inject annotation on a constructor instead.

### Testing instructions

- Make sure you are using a store which is eligible for in-person payments
1. Open detail of an unpaid order in USD
2. Tap on Collect Payment
3. Connect to a reader
4. Wait until the app prompts you to insert/tap/swipe your card
5. Notice the currency symbol at the top of the dialog is "$"

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt [hunk 1]
```diff
@@ -341,8 +342,7 @@ class CardReaderPaymentViewModel
             selectedSite.get().name.orEmpty()
         )
 
-    // TODO cardreader don't hardcode currency symbol ($)
-    private fun Order.getAmountLabel() = "$$total"
+    private fun Order.getAmountLabel() = "${currencyFormatter.getCurrencySymbol(this.currency)}$total"
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt]: For future reference, we might want to add a TODO here stating that some non-US currencies put the currency symbol to the right.

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentViewModel.kt]: ohh, great point.. I'll fix it in this PR.

**Comment by malinajirka** [comment]
@nbradbury Thanks for the first round of the review. I've fixed the currency position - based on the locale the currency is displayed in the correct format at the right position eg. $1000 with US locale but 1000 $US in FR locale.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/CurrencyFormatter.kt [hunk 2]
```diff
@@ -100,4 +110,14 @@ class CurrencyFormatter(private val wcStore: WooCommerceStore, private val selec
     fun buildBigDecimalFormatter(currencyCode: String) = { amount: BigDecimal ->
         formatCurrency(amount, currencyCode, true)
     }
+
+    /**
+     * Returns formatted amount with currency symbol - eg. $113.5 for EN/USD or 113,5€ for FR/EUR.
+     */
+    fun formatAmountWithCurrency(currencyCode: String, amount: Double): String {
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/CurrencyFormatter.kt:117]: 👍

**Review by nbradbury — APPROVED** [review]
Looks great! :shipit:

**Pull request merged**